<launch>
	<arg name="robot_name"/>
	<arg name="init_pose"/>
	<arg name="base"/>
	<arg name="stacks"/>
	<arg name="3d_sensor"/>

	<arg name="urdf_file" default="$(find xacro)/xacro.py '$(find turtlebot_description)/robots/$(arg base)_$(arg stacks)_$(arg 3d_sensor).urdf.xacro'" /> 
	<param name="robot_description" command="$(arg urdf_file)" />


	<node name="spawn_turtlebot_model" pkg="gazebo_ros" type="spawn_model"
		args="$(optenv ROBOT_INITIAL_POSE) -unpause -urdf -param robot_description -model $(arg robot_name) $(arg init_pose)"/>

    <node pkg="robot_state_publisher" type="robot_state_publisher" name="robot_state_publisher" output="screen"/>

    <!-- Velocity muxer -->
	<node pkg="nodelet" type="nodelet" name="mobile_base_nodelet_manager" args="manager"/>
	<node pkg="nodelet" type="nodelet" name="cmd_vel_mux" args="load yocs_cmd_vel_mux/CmdVelMuxNodelet mobile_base_nodelet_manager">
		<param name="yaml_cfg_file" value="$(find turtlebot_bringup)/param/mux.yaml" />
		<remap from="cmd_vel_mux/output" to="mobile_base/commands/velocity"/>
	</node>
	
	<!-- The odometry estimator, throttling, fake laser etc. go here -->
	<!-- All the stuff as from usual robot launch file -->

	<!-- Fake laser -->
	<node pkg="nodelet" type="nodelet" name="laserscan_nodelet_manager" args="manager"/>
	<node pkg="nodelet" type="nodelet" name="depthimage_to_laserscan"
		args="load depthimage_to_laserscan/DepthImageToLaserScanNodelet laserscan_nodelet_manager">
		<param name="scan_height" value="10"/>
		<param name="output_frame_id" value="camera_depth_frame"/>
		<param name="range_min" value="0.45"/>
		<remap from="image" to="camera/depth/image_raw"/>
		<remap from="scan" to="scan"/>
	</node>

</launch>
